src/provider/rpc.ts

```diff
@@ -231,14 +232,37 @@ export class RpcProvider implements ProviderInterface {
       throw Error('getContractVersion require contractAddress or classHash');
     }
 
+    // Take the opportunity of class reading, to get the name of the message signature verification name
+    // or an empty string in case of proxy.
+    let messageVerifFunctionName: string | undefined;
+    if ('signatureVerifFunctionName' in this) {
+      const parser = createAbiParser(contractClass.abi);
+      const parsedAbi = parser.getLegacyFormat();
+      const functionsList = parsedAbi.filter((abiElement) => abiElement.type === 'function');
+      if (functionsList) {
```

Comment:
This conditional can be removed, it will always be true. Or it could be modified to check the array length.

Reply:
```suggestion
      if (functionsList.length > 0) {
```

Reply:
Something like that?